core/src/main/java/io/lionweb/lioncore/java/serialization/NodePopulator.java

```diff
@@ -0,0 +1,116 @@
+                  containment,
+                  "Unable to resolve containment "
+                      + serializedContainmentValue.getMetaPointer()
+                      + " in concept "
+                      + concept);
+              Objects.requireNonNull(
+                  serializedContainmentValue.getValue(),
+                  "The containment value should not be null");
+              List<ClassifierInstance<?>> deserializedValue =
+                  serializedContainmentValue.getValue().stream()
+                      .map(childNodeID -> classifierInstanceResolver.strictlyResolve(childNodeID))
+                      .collect(Collectors.toList());
+              if (!Objects.equals(deserializedValue, node.getChildren(containment))) {
+                deserializedValue.forEach(child -> node.addChild(containment, (Node) child));
+              }
+            });
+  }
+
+  private void populateNodeReferences(
+      ClassifierInstance<?> node, SerializedClassifierInstance serializedClassifierInstance) {
+    Classifier<?> concept = node.getClassifier();
+    // TODO resolve references to Nodes in different models
+    serializedClassifierInstance
+        .getReferences()
+        .forEach(
+            serializedReferenceValue -> {
+              Reference reference =
+                  concept.getReferenceByMetaPointer(serializedReferenceValue.getMetaPointer());
+              if (reference == null) {
+                throw new IllegalStateException(
+                    "Unable to solve reference "
+                        + serializedReferenceValue.getMetaPointer()
+                        + ". Concept "
+                        + concept
+                        + ". SerializedNode "
+                        + serializedClassifierInstance);
+              }
+              serializedReferenceValue
+                  .getValue()
+                  .forEach(
+                      entry -> {
+                        Node referred =
+                            (Node) classifierInstanceResolver.resolve(entry.getReference());
+                        if (entry.getReference() != null && referred == null) {
+                          switch (jsonSerialization.getUnavailableReferenceTargetPolicy()) {
+                            case NULL_REFERENCES:
+                              referred = null;
+                              break;
+                            case PROXY_NODES:
+                              referred = deserializationStatus.createProxy(entry.getReference());
+                              break;
+                            case THROW_ERROR:
+                              throw new DeserializationException(
+                                  "Unable to resolve reference to "
+                                      + entry.getReference()
+                                      + " for feature "
+                                      + serializedReferenceValue.getMetaPointer());
+                            default:
```

Comment:
We don't need this, and it's actually harmful: If we would ever add a new literal to the enum, the compiler would tell us.